network/gossip/libp2p/topology/topicAwareTopology.go

```diff
@@ -0,0 +1,84 @@
+package topology
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/state/protocol"
+)
+
+// TopicAwareTopology is a deterministic topology mapping that creates a connected graph component among the nodes
```

Comment:
🤔  would `TopicBasedTopology` be a better name?

Reply:
Applied in https://github.com/onflow/flow-go/pull/83/commits/98e1433fa8bd08a91635d9d00a2e35fda90863eb.